sklearn/linear_model/least_angle.py

```diff
@@ -1114,59 +1119,60 @@ def _lars_path_residues(X_train, y_train, X_test, y_test, Gram=None,
     """Compute the residues on left-out data for a full LARS path
 
     Parameters
-    ----------
-    X_train : array, shape (n_samples, n_features)
+    -----------
+    X_train : array-like of shape (n_samples, n_features)
         The data to fit the LARS on
 
-    y_train : array, shape (n_samples)
+    y_train : array-like of shape (n_samples,)
         The target variable to fit LARS on
 
-    X_test : array, shape (n_samples, n_features)
+    X_test : array-like of shape (n_samples, n_features)
         The data to compute the residues on
 
-    y_test : array, shape (n_samples)
+    y_test : array-like of shape (n_samples,)
         The target variable to compute the residues on
 
-    Gram : None, 'auto', array, shape (n_features, n_features), optional
+    Gram : None, 'auto' or array-like of shape (n_features, n_features), \
+            default=None
         Precomputed Gram matrix (X' * X), if ``'auto'``, the Gram
         matrix is precomputed from the given X, if there are more samples
         than features
 
-    copy : boolean, optional
+    copy : bool, default=True
         Whether X_train, X_test, y_train and y_test should be copied;
         if False, they may be overwritten.
 
-    method : 'lar' | 'lasso'
+    method : {'lar' , 'lasso'}, default='lar'
         Specifies the returned model. Select ``'lar'`` for Least Angle
         Regression, ``'lasso'`` for the Lasso.
 
-    verbose : integer, optional
+    verbose : bool or int, default=False
         Sets the amount of verbosity
 
-    fit_intercept : boolean
+    fit_intercept : bool, default=True
         whether to calculate the intercept for this model. If set
         to false, no intercept will be used in calculations
         (e.g. data is expected to be already centered).
 
-    positive : boolean (default=False)
+    positive : bool, default=False
         Restrict coefficients to be >= 0. Be aware that you might want to
         remove fit_intercept which is set True by default.
         See reservations for using this option in combination with method
         'lasso' for expected small values of alpha in the doc of LassoLarsCV
         and LassoLarsIC.
 
-    normalize : boolean, optional, default True
+    normalize : bool, default=True
         This parameter is ignored when ``fit_intercept`` is set to False.
         If True, the regressors X will be normalized before regression by
         subtracting the mean and dividing by the l2-norm.
         If you wish to standardize, please use
         :class:`sklearn.preprocessing.StandardScaler` before calling ``fit``
         on an estimator with ``normalize=False``.
 
-    max_iter : integer, optional
+    max_iter : int, default=500
         Maximum number of iterations to perform.
 
-    eps : float, optional
+    eps : float, default=np.finfo(np.float).eps
```

Comment:
I think this is one of the few cases where we could describe the default in the description instead.

E.g.

```
eps : float, optional
    [...]. By default, ``np.finfo(np.float).eps`` is used

Reply:
@NicolasHug and use `optional` ?

Reply:
Yes, since the default isn't mentioned on the first line I think it's worth clarifying that the parameter is still optional